tests/integration/pilot/tunneling_test.go

```diff
@@ -220,60 +220,27 @@ func applyForwardProxyConfigMaps(ctx framework.TestContext, externalNs string) {
 		ctx.Fatalf("failed to generate private key and certificate: %s", err)
 	}
 
-	cfgMap := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "external-forward-proxy-config",
-		},
-		Data: map[string]string{
-			"envoy.yaml":                      bootstrapYaml,
-			"external-forward-proxy-key.pem":  key,
-			"external-forward-proxy-cert.pem": crt,
-		},
-	}
-	if _, err := kubeClient.CoreV1().ConfigMaps(externalNs).Create(context.TODO(), cfgMap, metav1.CreateOptions{}); err != nil {
-		ctx.Fatalf("failed to create config map external-forward-proxy-config: %s", err)
-		if kerrors.IsAlreadyExists(err) {
-			if _, err := kubeClient.CoreV1().ConfigMaps(externalNs).Update(context.TODO(), cfgMap, metav1.UpdateOptions{}); err != nil {
-				ctx.Fatalf("failed to update config map external-forward-proxy-config: %s", err)
-			}
-		} else {
-			ctx.Fatalf("failed to create config map external-forward-proxy-config: %s", err)
-		}
+	templateParams := map[string]any{
+		"envoyYaml": bootstrapYaml,
+		"keyPem":    key,
+		"certPem":   crt,
 	}
+	ctx.ConfigIstio().EvalFile(externalNs, templateParams, forwardProxyConfigMapFile).ApplyOrFail(ctx)
 }
 
 func applyForwardProxyService(ctx framework.TestContext, externalNs string) {
-	kubeClient := ctx.Clusters().Default().Kube()
-
-	svc := &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "external-forward-proxy",
-		},
-		Spec: corev1.ServiceSpec{
-			Ports: []corev1.ServicePort{},
-			Selector: map[string]string{
-				"app": "external-forward-proxy",
-			},
-		},
-	}
+	var servicePorts []corev1.ServicePort
 	for i, cfg := range forwardProxyConfigurations {
-		svc.Spec.Ports = append(svc.Spec.Ports, corev1.ServicePort{
+		servicePorts = append(servicePorts, corev1.ServicePort{
 			Name:       fmt.Sprintf("%s-%d", selectPortName(cfg.HTTPVersion), i),
 			Port:       int32(cfg.Port),
 			TargetPort: intstr.FromInt(int(cfg.Port)),
-			Protocol:   corev1.ProtocolTCP,
 		})
 	}
-	if _, err := kubeClient.CoreV1().Services(externalNs).Create(context.TODO(), svc, metav1.CreateOptions{}); err != nil {
-		ctx.Fatalf("failed to create service external-forward-proxy: %s", err)
```

Comment:
This is also potential cause of failures. It didn't failed so far, but the if statement below is unreachable.